charts/cortx/templates/_helpers.tpl

```diff
@@ -269,3 +269,15 @@ Return the Motr-client endpoint port
 {{- define "cortx.client.motrClientPort" -}}
 21501
 {{- end -}}
+
+{{/*
+Create a URL CORTX Confstore URL
+*/}}
+{{- define "cortx.confstore.url" -}}
+{{- if .Values.consul.enabled -}}
+{{- printf "consul://%s-consul-server:8500/conf" (include "cortx.fullname" .) -}}
+{{- else -}}
+{{/* FIXME: how do we get the external service name? */}}
+{{- printf "yaml:///etc/cortx/cluster.conf" -}}
+{{- end -}}
```

Comment:
I'm not sure what to do here. We've never validated external Consul configurations, so this works as long as the CORTX images support backwards compatibility.

The consul endpoints provided by the user are just a list of tcp and http endpoints. The URL in line 278 is neither of those, but it corresponds to the http endpoint, so presumably we could try to search for it. https://github.com/Seagate/cortx-prvsnr/blob/a98e8cafee57e06a75626c31e02ddf4398724e3e/conf/solution/config.yaml.sample#L7

Reply:
I think for now this is fine to tag it as-is with a future TODO. There aren't explicit requirements to support external Consul deployments or a Consul deployment that we (as the cortx-k8s deployer) don't control the service name of. So for the time being, what is written above is valid.

Once requirements change to support explicit external Consul deployments, then we can revisit that as a Helm Chart input and pulling that in directly via values.yaml.